spec/requests/create_resource_spec.rb

```diff
@@ -207,6 +207,24 @@
       end
     end
 
+    context 'when the registration request unexpectedly fails' do
+      before do
+        allow(Dor::Services::Client.objects).to receive(:register)
+          .and_raise(Dor::Services::Client::UnexpectedResponse,
+                     "Conflict: 409 (An object with the source ID 'abcd:1234' already registered)")
+      end
+
+      let(:error) { JSON.parse(response.body)['errors'][0] }
+
+      it 'returns an error response' do
+        post '/v1/resources',
+             params: request,
+             headers: { 'Content-Type' => 'application/json', 'Authorization' => "Bearer #{jwt}" }
+        expect(response).to have_http_status(:bad_request)
```

Comment:
This status should be conflict as well.

Reply:
Do you mean the `:bad_request`? The intention (I think)  in the `build_error` method is to follow the https://jsonapi.org/format/#errors suggestions, the over-all response has a more general error of `400` because there could be multiple errors with different error codes in the JSON `error` array.

Reply:
While there could be more than one code in the array according to JSON-API, this is never true in our case.  We just want this to act as a simple proxy to dor-service-app and return the same error code that sdr-api received.

Reply:
👍 Just wanted to confirm that assumption. Pushed commit that sets status to error code if it exists.